src/main/java/gmailgdogra/ReadXlsxService.java

```diff
@@ -0,0 +1,24 @@
+package gmailgdogra;
+
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+
+@Service
+public class ReadXlsxService {
+
+    public static String EXCEL_FILE_TYPE = "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet";
+
+    public static List<SwipeRecord> parse(InputStream inputStream) throws IOException {
```

Comment:
Where do you actually catch this IOException? Also, ask around what is the best practic: return an empty list or propagate the exception.

Reply:
I was catching this in RestController
```
catch (Exception e) {
                return ResponseEntity.status(HttpStatus.EXPECTATION_FAILED)
                        .body(new ResponseMessage("Unable to parse file: " + file.getName()));
            }
```
And there is a test for this exception being thrown